4_predict/src/train_models.R

```diff
@@ -555,6 +555,228 @@ filter_rows_date <- function(attrs, start_date){
 
 
 #SHAP values
+setup_shap_data <- function(data, seasonal = TRUE, physio = TRUE, lulc = TRUE,
+                            lulc_prop = 0.75, lulc_data = NULL, physio_data = NULL){
+  #' @description prepares data for SHAP computation based on the provided split
+  #'
+  #' @param data the dataset with predictor attributes
+  #' @param seasonal logical for splitting data by water year season
+  #' @param physio logical for splitting data by physiographic region
+  #' @param lulc logical for splitting data by lulc, as defined into 2 groups:
+  #' high urban land cover and high forest land cover
+  #' @param lulc_prop threshold defining "high" proportion land cover on [0,1]
+  #' @param lulc_data land cover data for total upstream urban and forest. Must
+  #' also have the PRMS_segid and Date columns
+  #' @param physio_data physiographic region data. 
+  #' Must also have the PRMS_segid and Date columns
+  #' 
+  #' @return Returns a list with the different data splits
+  
+  data_lst <- list()
+  data_lst_names <- c()
+  
+  if(seasonal){
+    #split data by water year season (4 elements in list)
+    data$months <- lubridate::month(data$Date)
+    OND <- filter(data, months %in% c(10,11,12)) %>%
```

Comment:
lubridate also has a function called `quarter` that could be useful for cases like this (for future reference, this looks good and seems like you're actually using these list names).